01_fib/alfredxiao/main.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func fibSeries(n int) []int {
+	if n >= 93 || n <= -93 {
+		panic("Numbers beyond 93 or -93 are not supported!")
+	}
+	var series []int
+	fa, fb := 0, 1
+	p, fn := 0, 0
+	for {
+		switch {
+		case p == 0:
+			fn = fa
+		case p == 1:
+			fn = fb
+		case p > 1:
+			fa, fb = fb, fa+fb
+			fn = fb
+		case p < 0:
+			fa, fb = fb-fa, fa
+			fn = fa
+		}
+		series = append(series, fn)
+		if p == n {
+			break
+		}
+		if n > 0 {
+			p++
+		} else if n < 0 {
+			p--
+		}
+	}
+	return series
```

Comment:
according to your comprehensive tests (well done!) it looks like this is functionally correct, however, it's a little hard to reason about imo and could be shortened, especially the append can be avoided by pre-allocating the needed slice size:

```go
	sign := 1
	if n < 0 {
		sign = -1
	}

	n1, n2 := 0, 1
	result := make([]int, n*sign)
	for i := 0; i < n*sign; i++ {
		n1, n2 = n2, n1+sign*n2
		result[i] = n1
	}
	return result
```
or even shorter if you prefer (need to check for 0 though):
```go
	result := make([]int, n*sign+1)
	result[0], result[1] = 0, 1
	for i := 2; i <= n*sign; i++ {
		result[i] = result[i-2] + result[i-1]*sign
	}
	return result[1:]

```

Reply:
Yeah, I know there must be a simpler version of this logic